### ISSUE TITLE
homebridge not starting due to sensibo

### ISSUE THREAD

**Issue by andrewjones141** [wailuen/homebridge-sensibo-sky#6]
having this isssue today where homebridge wont start, seems sensibo plugin is having issues:
pi@raspberrypi:~ $ homebridge
*** WARNING *** The program 'node' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node>
*** WARNING *** The program 'node' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=node&f=DNSServiceRegister>
[7/2/2017, 11:56:43 AM] Loaded plugin: homebridge-homeassistant
[7/2/2017, 11:56:44 AM] Registering platform 'homebridge-homeassistant.HomeAssistant'
[7/2/2017, 11:56:44 AM] ---
[7/2/2017, 11:56:44 AM] Loaded plugin: homebridge-platform-wemo
[7/2/2017, 11:56:44 AM] Registering platform 'homebridge-platform-wemo.BelkinWeMo'
[7/2/2017, 11:56:44 AM] ---
[7/2/2017, 11:56:44 AM] Loaded plugin: homebridge-sensibo-sky
[7/2/2017, 11:56:45 AM] Registering platform 'homebridge-sensibo-sky.SensiboSky'
[7/2/2017, 11:56:45 AM] ---
[7/2/2017, 11:56:45 AM] Loaded plugin: homebridge-zway
[7/2/2017, 11:56:45 AM] Registering accessory 'homebridge-zway.ZWayServer'
[7/2/2017, 11:56:45 AM] Registering platform 'homebridge-zway.ZWayServer'
[7/2/2017, 11:56:45 AM] ---
[7/2/2017, 11:56:45 AM] Loaded config.json with 0 accessories and 3 platforms.
[7/2/2017, 11:56:45 AM] ---
[7/2/2017, 11:56:45 AM] Loading 3 platforms...
[7/2/2017, 11:56:45 AM] [WeMo Platform] Initializing BelkinWeMo platform...
[7/2/2017, 11:56:45 AM] Initializing ZWayServer platform...
[7/2/2017, 11:56:45 AM] [Sensibo] Initializing SensiboSky platform...
[7/2/2017, 11:56:45 AM] [Sensibo] Fetching Sensibo devices...
[7/2/2017, 11:56:46 AM] [WeMo Platform] Online: garage light [94103E4204D8]
[7/2/2017, 11:56:46 AM] Initializing platform accessory 'Aeotec Temperature (#6)'...
/usr/local/lib/node_modules/homebridge-sensibo-sky/index.js:57
                                for (var i = 0; i < devices.length; i++) {
                                                           ^

TypeError: Cannot read property 'length' of undefined
    at /usr/local/lib/node_modules/homebridge-sensibo-sky/index.js:57:32
    at /usr/local/lib/node_modules/homebridge-sensibo-sky/lib/sensiboapi.js:100:5
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-sensibo-sky/lib/sensiboapi.js:50:21)
    at emitNone (events.js:73:20)
    at IncomingMessage.emit (events.js:167:7)
    at endReadableNT (_stream_readable.js:906:12)
    at nextTickCallbackWith2Args (node.js:455:9)
    at process._tickCallback (node.js:369:17)

**Comment by andrewjones141** [comment]
just tried this again on a fresh install as i am setting one up for a friend, and get the same error. perhaps the api has changed somehow? as i did have it working fine a few days ago

**Comment by wailuen** [comment]
Same error. Looks like nothing is returned from Sensibo server. Did you get the correct API key from Sensibo?

**Comment by wailuen** [comment]
Added check for NULL devices return from sensibo server. V0.2.9. Please check API key or internet connection.

**Comment by andrewjones141** [comment]
hey, i justgot a chance to test this, and turns out my API key was incorrect. sorry for troubling you, but at-least after the update the log told me what the real problem was.

Thanks,

**Comment by Cloudore** [comment]
> Added check for NULL devices return from sensibo server. V0.2.9. Please check API key or internet connection.

Sorry to keep pestering you with these @wailuen 

This update made it so every time my Sensibo hub couldn't reach the server, be it because the light or internet was out, or simply because my hub didn't want to cooperate, every Sensibo device in my Home app(3 ACs, 3Fans, 3 Humidity Sensors) would disappear. And then when they started responding again, they would reappear all in the default room with default names. This annoyed me because it happened like 5 times until I realized what was happening.

I went ahead and reverted on my files the changes made on this commit, but though you would like to know.

Thanks as always!
